src/layer.js

```diff
@@ -148,7 +141,24 @@ export class MapLayer extends HTMLElement {
           this.opacity = newValue;
         }
       break;
+      case 'src':
+        if (oldValue !== newValue && this._layer) {
+          this._reload();
+          // the original inline content will not be removed
+          // but has NO EFFECT and works as a fallback
+        }
+    }
+  }
+  // re-load the layer element when the src attribute is changed
+  _reload() {
+    let oldOpacity = this.opacity;
+    // go through the same sequence as if the layer had been removed from
+    // the DOM and re-attached with a new URL source.
+    this.disconnectedCallback();
```

Comment:
I think it's kind of odd to call disconnectedCallback ourselves.  This should be done by the browser; maybe what we need is a _privateMethodThatImpliesTheProcessingThatOccursDuringDisconnectedCallback, and call that from disconnectedCallback, and here.